tests/test_ilastik/test_applets/base/test_serializerUtils.py

```diff
@@ -0,0 +1,552 @@
+    with pytest.raises(ValueError, match="Start indices for slicing must be integer, got `None`."):
+        _ = slicingToString(slicing)
+
+
+@pytest.mark.parametrize(
+    "slicing",
+    [
+        (slice(0, None),),
+        (slice(0, None), slice(5, 42, None)),
+    ],
+)
+def test_slicingToString_stop_none_raises(slicing):
+    with pytest.raises(ValueError, match="Stop indices for slicing must be integer, got `None`"):
+        _ = slicingToString(slicing)
+
+
+@pytest.mark.parametrize(
+    "slice_string,expected_slicing",
+    [
+        (b"[0:1]", (slice(0, 1),)),
+        (b"[0:1,5:42]", (slice(0, 1), slice(5, 42))),
+        ("[0:1,5:42]", (slice(0, 1), slice(5, 42))),
+    ],
+)
+def test_stringToSlicing(slice_string: Union[bytes, str], expected_slicing: Tuple[slice, ...]):
+    assert stringToSlicing(slice_string) == expected_slicing
+
+
+@pytest.mark.parametrize(
+    "slice_string",
+    [
+        b"[0:None]",
+        b"[None:1,5:42]",
+        "[0:1:5,5:42]",
+    ],
+)
+def test_stringToSlicing_raises(slice_string: Union[bytes, str]):
+    with pytest.raises(ValueError):
+        _ = stringToSlicing(slice_string)
+
+
+def test_deserialize_string_from_h5(empty_in_memory_project_file: h5py.File):
+    test_string = "this is a test string"
+    ds = empty_in_memory_project_file.create_dataset("test", data=test_string.encode("utf-8"))
+
+    assert deserialize_string_from_h5(ds) == test_string
+
+
+def test_deserialize_classifier(empty_in_memory_project_file: h5py.File):
+    classifier_bytes = b"clazyflow.classifiers.vigraRfLazyflowClassifier\nVigraRfLazyflowClassifier\np0\n."
+    expected_submodule = "vigraRfLazyflowClassifier"
+    expected_type = "VigraRfLazyflowClassifier"
+    ds = empty_in_memory_project_file.create_dataset("classifier_type", data=classifier_bytes)
+
+    cl_info = deserialize_legacy_classifier_type_info(ds)
+
+    assert cl_info.submodule_name == expected_submodule
+    assert cl_info.type_name == expected_type
+
+    assert issubclass(cl_info.classifier_type, VigraRfLazyflowClassifier)
+
+
+@pytest.mark.parametrize(
+    "classifier_bytes",
+    [
+        b"clazyflow.class.vigraRfLazyflowClassifier\nVigraRfLazyflowClassifierFactory\np0\n.",
+        b"csome.other_submodule.classifiers.vigraRfLazyflowClassifier\nVigraRfLazyflowClassifierFactory\np0\n.",
+        b"clazyflow.classifiers.sneakyVigraRfLazyflowClassifier\nVigraRfLazyflowClassifierFactory\np0\n.",
+        b"clazyflow.classifiers.vigraRfLazyflowClassifier\nSneakyVigraRfLazyflowClassifierFactory\np0\n.",
+        b"random.",
+    ],
+)
+def test_deserialize_classifier_raises(empty_in_memory_project_file: h5py.File, classifier_bytes: bytes):
+    ds = empty_in_memory_project_file.create_dataset("classifier_type", data=classifier_bytes)
+    with pytest.raises(ValueError):
+        _ = deserialize_legacy_classifier_type_info(ds)
+
+
+@pytest.mark.parametrize(
+    "classifier_type, c_args, c_kwargs, expected_info",
+    [
+        (
```

Comment:
Apparently this parameter collection is duplicated in lines 244+, could extract them into a variable. Or merge the two tests. Or remove the tests for the private functions altogether since they are covered by the tests on the public functions...

Similarly for `test_deserialize_classifier_factory_details` and `test_deserialize_legacy_classifier_factory`.

Reply:
I looked at it with Fynn and we decided to keep the duplicated parameters.